xcode/Subconscious/Shared/Components/Common/Transclude/TranscludeView.swift

```diff
@@ -2,41 +2,95 @@
 //  TranscludeView.swift
 //  Subconscious
 //
-//  Created by Gordon Brander on 8/23/22.
+//  Created by Gordon Brander on 2/22/23.
 //
 
 import SwiftUI
 
+struct ExcerptView: View {
+    var excerpt: String
+    var lineLimit: Int = 5
+    var spacing: CGFloat = AppTheme.unit
+    var excerptLines: [EnumeratedSequence<[String.SubSequence]>.Element] {
```

Comment:
Instead of flattening Subtext to a string and then splitting it again, we should have a method:

```
Subtext.truncate(maxBlocks: Int) -> Subtext
```

And hold a reference to this on the `EntryStub`. That way we're doing less work.

Reply:
I'm not 100% clear on how this would actually work? We create the excerpt and save it to the DB as a string and then load that string to populate transclusions and backlinks, so it _has_ to be flatted to a string before we get access to it.

I considered re-parsing as Subtext on load but that seems _even more_ wasteful?

Reply:
My mistake. You are correct.